src/test/java/org/opensearch/knn/integ/PainlessScriptIT.java

```diff
@@ -161,6 +161,42 @@ public void testL2ScriptScoreFails() throws Exception {
         deleteKNNIndex(INDEX_NAME);
     }
 
+    public void testCosineSimilarityScriptFields() throws Exception {
```

Comment:
1. Wouldn't it be better to have these test in a separate file?
2. Shouldn't we test it for other similarity functions?

Reply:
For 1) it is better keep them together since the functionality is same, and, it is within the scope of painless scripting.
2) From feature point of you, we added 6 contexts, testing all methods across 6 contexts is overkill, since from implementation point of view, either all works or none.

Reply:
I understand that but you know that we have painless scripts score tests for all space type. That is why maybe separating them could be better and mention that the if painless script score works, painless script_fields will work and no need to add tests for all space type. 

It would be easier to understand that there are two different thing inside painless script(script score and script field) if we have them in a separate file. Maybe we can change current PainlessScriptIT to PainlessScriptScoreIT as well?

Reply:
Renamed class and moved test method to its own class